### PR TITLE
Release 5.8.0

### PR DESCRIPTION
**Pull request by mgreisen**
## Summary of the Pull Request
Release 5.8.0

### CODE REVIEW

**Review by tevoinea — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,34 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+### 5.8.0
+### Added
+* Service: Add correct routes and auth to agent C# functions. [#2109](https://github.com/microsoft/onefuzz/pull/2109)
+* Service: Port `agent_registration` to C#. [#2107](https://github.com/microsoft/onefuzz/pull/2107)
+* Agent: Add the dotnet coverage task. [#2062](https://github.com/microsoft/onefuzz/pull/2062)
+* Agent: Added multiple ways to specify the dotnet Libfuzzer target. [#2136](https://github.com/microsoft/onefuzz/pull/2136)
+* Agent: Add logging to the LibFuzzer Dotnet loader. [#2141](https://github.com/microsoft/onefuzz/pull/2141)
```
Comment (ranweiler) [inline CHANGELOG.md:13]:
```suggestion
* Agent: Add logging to LibFuzzerDotnetLoader. [#2141](https://github.com/microsoft/onefuzz/pull/2141)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,34 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+### 5.8.0
+### Added
+* Service: Add correct routes and auth to agent C# functions. [#2109](https://github.com/microsoft/onefuzz/pull/2109)
+* Service: Port `agent_registration` to C#. [#2107](https://github.com/microsoft/onefuzz/pull/2107)
+* Agent: Add the dotnet coverage task. [#2062](https://github.com/microsoft/onefuzz/pull/2062)
+* Agent: Added multiple ways to specify the dotnet Libfuzzer target. [#2136](https://github.com/microsoft/onefuzz/pull/2136)
```
Comment (ranweiler) [inline CHANGELOG.md:12]:
```suggestion
* Agent: Add multiple ways to specify LibFuzzerDotnetLoader targets. [#2136](https://github.com/microsoft/onefuzz/pull/2136)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,34 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+### 5.8.0
+### Added
+* Service: Add correct routes and auth to agent C# functions. [#2109](https://github.com/microsoft/onefuzz/pull/2109)
+* Service: Port `agent_registration` to C#. [#2107](https://github.com/microsoft/onefuzz/pull/2107)
+* Agent: Add the dotnet coverage task. [#2062](https://github.com/microsoft/onefuzz/pull/2062)
```
Comment (ranweiler) [inline CHANGELOG.md:11]:
```suggestion
* Agent: Add the `dotnet_coverage` task. [#2062](https://github.com/microsoft/onefuzz/pull/2062)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,34 @@ All notable changes to this project will be documented in this file.
...
+* Documentation: Added documentation for LibFuzzerDotnetLoader. [#2142](https://github.com/microsoft/onefuzz/pull/2142)
+
+### Changed
+* Service: Add caching to C# storage implementation so repeated queries do not get throttled. [#2102](https://github.com/microsoft/onefuzz/pull/2102)
+* Service: Remove unused poolname validation. [#2094](https://github.com/microsoft/onefuzz/pull/2094)
+* Service: Make the hostbuilder async in C#. [#2122](https://github.com/microsoft/onefuzz/pull/2122)
+* Service: Include State filter when searching for expired tasks and jobs. [#2138](https://github.com/microsoft/onefuzz/pull/2138)
+* Service: Updated the scaling policy for the App Functions. [#2140](https://github.com/microsoft/onefuzz/pull/2140)
+* Agent/Supervisor/Proxy: Updated multiple third-party Rust dependencies. [#2022](https://github.com/microsoft/onefuzz/pull/2022)
+* Agent: Remove MacOS support. [#2134](https://github.com/microsoft/onefuzz/pull/2134), [#2135](https://github.com/microsoft/onefuzz/pull/2135)
```
Comment (ranweiler) [inline CHANGELOG.md:23]:
```suggestion
* Agent: Remove incomplete macOS support. [#2134](https://github.com/microsoft/onefuzz/pull/2134), [#2135](https://github.com/microsoft/onefuzz/pull/2135)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,34 @@ All notable changes to this project will be documented in this file.
...
+
+### Fixed
+* Service: Fix reported TLS errors. [#2087](https://github.com/microsoft/onefuzz/pull/2087)
+* Service: Change the `upload_file` method to use the Azcopy command by default for robustness and fall back to the Azure Python SDK implementation if needed. This also addresses issues where low bandwidth connections timeout due to not being able to handle multiple concurrent upload streams.  [#1556](https://github.com/microsoft/onefuzz/pull/1556)
+* Service: Update the log Sas URL to last as long as the job duration. [#2116](https://github.com/microsoft/onefuzz/pull/2116), [#2121](https://github.com/microsoft/onefuzz/pull/2121)
```
Comment (ranweiler) [inline CHANGELOG.md:28]:
```suggestion
* Service: Update the log SAS URL to last as long as the job duration. [#2116](https://github.com/microsoft/onefuzz/pull/2116), [#2121](https://github.com/microsoft/onefuzz/pull/2121)
```

**Review by ranweiler — APPROVED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,34 @@ All notable changes to this project will be documented in this file.
...
+* Agent: Add the `dotnet_coverage` task. [#2062](https://github.com/microsoft/onefuzz/pull/2062)
+* Agent: Add multiple ways to specify LibFuzzerDotnetLoader targets. [#2136](https://github.com/microsoft/onefuzz/pull/2136)
+* Agent: Add logging to LibFuzzerDotnetLoader. [#2141](https://github.com/microsoft/onefuzz/pull/2141)
+* Documentation: Added documentation for LibFuzzerDotnetLoader. [#2142](https://github.com/microsoft/onefuzz/pull/2142)
+
+### Changed
+* Service: Add caching to C# storage implementation so repeated queries do not get throttled. [#2102](https://github.com/microsoft/onefuzz/pull/2102)
+* Service: Remove unused poolname validation. [#2094](https://github.com/microsoft/onefuzz/pull/2094)
+* Service: Make the hostbuilder async in C#. [#2122](https://github.com/microsoft/onefuzz/pull/2122)
+* Service: Include State filter when searching for expired tasks and jobs. [#2138](https://github.com/microsoft/onefuzz/pull/2138)
```
Comment (chkeita) [inline CHANGELOG.md:20]: The state filter change  should be in the fixed section

**Review by chkeita — APPROVED** [review]

File: CHANGELOG.md (hunk 6 above)
Comment (mgreisen) [reply CHANGELOG.md]: Updated

**Pull request merged**